### ISSUE TITLE
Since version 1.1.326 pyright allows to assign the wrong type to a field of an sqlalchemy ORM class

### ISSUE THREAD

**Issue by arjenzorgdoc** [microsoft/pyright#6089]
**Describe the bug**
Since version 1.1.326 pyright allows to assign the wrong type to a field of an sqlalchemy ORM class:

```python
from sqlalchemy.orm import DeclarativeBase, Mapped, mapped_column


class Base(DeclarativeBase):
    pass


class Widget(Base):
    __tablename__ = "widget"
    id: Mapped[int] = mapped_column(primary_key=True)


w = Widget()
w.id = "test"  # this should be an error!
```

```shell
$ pip install sqlalchemy==2.0.21
[...]
$ for e in pyright@1.1.{330,326,325}; do echo ========= $e ======== ; npx $e model.py; done
========= pyright@1.1.330 ========
0 errors, 0 warnings, 0 informations 
========= pyright@1.1.326 ========
0 errors, 0 warnings, 0 informations 
========= pyright@1.1.325 ========
/.../src/pyright-bug/model.py
  /.../src/pyright-bug/model.py:14:8 - error: Cannot assign member "id" for type "Widget"
    Type "Literal['test']" cannot be assigned to type "SQLCoreOperations[int] | int"
      "Literal['test']" is incompatible with "SQLCoreOperations[int]"
      "Literal['test']" is incompatible with "int" (reportGeneralTypeIssues)
1 error, 0 warnings, 0 informations 
```

**Comment by erictraut** [comment]
Thanks for the bug report. This regression was introduced as part of [this bug fix](https://github.com/microsoft/pyright/issues/5897). The reason it happens with your example is because `DeclarativeBase` derives from a class that has a custom metaclass that supplies a `__setattr__` method. Pyright was incorrectly handling this case.

This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.331, which I just published. It will also be included in a future release of pylance.
